lib/MultiCompiler.js

```diff
@@ -143,10 +143,14 @@ MultiCompiler.prototype.watch = function(watchOptions, handler) {
 				handler(err);
 			if(stats) {
 				allStats[compilerIdx] = stats;
-				compilerStatus[compilerIdx] = true;
+				compilerStatus[compilerIdx] = stats.hash;
 				if(compilerStatus.every(Boolean)) {
-					var multiStats = new MultiStats(allStats);
+					var freshStats = allStats.filter(function(s) {
+						return compilerStatus.indexOf(s.hash) > -1
+					});
+					var multiStats = new MultiStats(freshStats);
 					handler(null, multiStats)
+					compilerStatus.fill(true);
```

Comment:
Could you move this line two line up? If someone doing weird stuff in the handler.